nautobot/dcim/api/views.py

```diff
@@ -251,10 +251,6 @@ class RackReservationViewSet(NautobotModelViewSet):
     serializer_class = serializers.RackReservationSerializer
     filterset_class = filters.RackReservationFilterSet
 
-    # Assign user from request
-    def perform_create(self, serializer):
-        serializer.save(user=self.request.user)
-
```

Comment:
I don't know why tests were passing before, but this was definitely wrong and causing test failures now. We already handle the case in `RackReservationSerializer.to_internal_value()` where `user` is omitted and we want to use `request.user`, but this was ensuring that on create we would *always* use `request.user` even if a specific other `user` was provided.

Reply:
This might be because of the change in default serializer depth to 0; `test_recreate_object_csv()` directly instantiates serializers and I think at depth 1 any nested serializers get flagged as read-only by DRF and so were skipped over in the data validation.